### PR TITLE
Wire up ask/answer components to the stage

### PR DESCRIPTION
**Pull request by paulkaplan**
screen shot 2017-11-08 at 9 53 12 am

### Resolves

_What Github issue does this resolve (please include link)?_

GUI part of ask/answer block implementation

### Proposed Changes

_Describe what this Pull Request does_

Listen to the `QUESTION` events, create and manage the question/answer inputs, and submit user input to the VM through the `ANSWER` event. 

hidden-ask

### CODE REVIEW

**Comment by paulkaplan** [comment]
@carljbowman I used the closest possible unicode checkmark character. Send me an asset for the checkmark (or the whole circle button) whenever you can and I can update it.

File: src/containers/stage.jsx [hunk 1]
```diff
@@ -47,12 +50,14 @@ class Stage extends React.Component {
         this.updateRect();
         this.renderer = new Renderer(this.canvas);
         this.props.vm.attachRenderer(this.renderer);
+        this.props.vm.runtime.addListener('QUESTION', this.questionListener);
```
[the remark this answers is not in the document]
Comment (rschamp) [reply src/containers/stage.jsx]: Another idea on this, maybe in the short term we should add a Runtime event, `BLOCK_EVENT`, which blocks packages can fire with arbitrary data? Then we can set up the echoed handler on the VM without knowledge of which blocks packages will use it.  So here we could add a listener:

```jsx
this.props.vm.addListener('BLOCK_EVENT', this.questionListener);
// ... snip
questionListener ({type, data}) {
    if (type !== 'QUESTION') return;
    // data is {question: question}...
}
```

**Comment by fsih** [comment]
I'm still reviewing this, but I wanted to mention this bug before you submit. If you click the "ask ... and wait" block and then switch tabs, then switch back, Scratch crashes.

**Comment by fsih** [comment]
Maybe-not-bugs or maybe separate bugs:
1. Should clicking on the script to stop it stop asking the question?
2. In 2.0, if the ask and answer block is in a sprite, the question comes from a say bubble from the sprite rather than appearing above the input box.
3. If you drop some blocks such as 'answer' and 'timer' into the text field, then you click on the ask and wait block, it doesn't glow.
image
4. Monitor for empty answer
image
5. Answer text goes under the checkmark
image
6. The queueing of questions works differently from 2.0
- Drag 3 'ask <answer> and wait' blocks to the workspace
- Click on all 3
image
Behavior in 2.0: The question is nothing, then your first answer, then your second ansewr
Behavior in 3.0: The question is nothing all 3 times
If you use timer, 3.0 uses time when you started the block, 2.0 uses time when it ran.
You can also do this across sprites (if 2 sprites want to ask questions at the same time)

**Comment by paulkaplan** [comment]
thanks for the thorough review @fsih. I fixed the crashing when changing tabs (72dbf9f and 
0e4b099) and the text going below the button (00d523a). The empty answer monitor would be fixed through scratch-blocks. Good catch on the queue behavior, could you file that case on vm?

File: src/lib/make-toolbox-xml.js [hunk 2]
```diff
@@ -336,12 +336,12 @@ const control = `
                 </shadow>
             </value>
         </block>
-        <block type="control_forever"/>
+        <block id="forever" type="control_forever"/>
```
Comment (fsih) [inline src/lib/make-toolbox-xml.js]: Does it make sense to add IDs to all blocks?

**Comment by paulkaplan** [comment]
@rschamp I'm going to file an issue about reaching into the runtime (we are doing in a few places) so we can handle those all together. I think there might be more solutions that come up when we consider all the places that do this at the same time.

**Pull request merged**